### PR TITLE
fixed goroutine leak, when ws connection reconnect, but fifo's channel no close.

### LINKED ISSUE #2018: goroutine leak
**What happened**:
The edge side will happen goroutine leak when the edgecore is often disconnected from the cloudhub communication

**What you expected to happen**:
When the edgecore is disconnected from the cloudhub, and edge side's connection will reconnection, but it not close fifo's channel, and it will cause  goroutine leak

**How to reproduce it (as minimally and precisely as possible)**:
Disconnect a little bit more often, and then watch if the goroutine number is growing all the time.

**Environment**:
- KubeEdge version(e.g. `cloudcore/edgecore --version`):  
   All version

**CloudSide Environment**:
- Hardware configuration (e.g. `lscpu`):
- OS (e.g. `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Go version (e.g. `go version`): 
- Others:

**EdgeSide Environment**:
- edgecore version (e.g. `edgecore --version`):
- Hardware configuration (e.g. `lscpu`):
- OS (e.g. `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Go version (e.g. `go version`): 
- Others:

### PR DESCRIPTION
**Pull request by loongify**
…l not close

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug

**What this PR does / why we need it**:
fixed goroutine leak: when the edgecore is disconnected from the cloudhub, and edge side's connection will reconnection, but it not close fifo's channel, and it will cause goroutine leak
**Which issue(s) this PR fixes**:

Fixes #2018

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: staging/src/github.com/kubeedge/viaduct/pkg/fifo/msgfifo.go [hunk 1]
```diff
@@ -43,3 +45,9 @@ func (f *MessageFifo) Get(msg *model.Message) error {
 	}
 	return nil
 }
+
+func (f *MessageFifo) Close() {
+	f.closeOnce.Do(func() {
+		close(f.fifo)
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/viaduct/pkg/fifo/msgfifo.go:51]: Why do we only close once? Looks like the fifo will be recreated every time reconnect.

Comment (loongify) [reply staging/src/github.com/kubeedge/viaduct/pkg/fifo/msgfifo.go:51]: Only same object close once, two different object will be closed again.

Comment (fisherxu) [reply staging/src/github.com/kubeedge/viaduct/pkg/fifo/msgfifo.go:51]: Okey :), so for the same object, is it necessary to use once?

Comment (loongify) [reply staging/src/github.com/kubeedge/viaduct/pkg/fifo/msgfifo.go:51]: yes, closing a closed channel will panic, so the channel should be closed once

Comment (fisherxu) [reply staging/src/github.com/kubeedge/viaduct/pkg/fifo/msgfifo.go:51]: Make sense! And would you like to split the second commit to another pr?

Comment (loongify) [reply staging/src/github.com/kubeedge/viaduct/pkg/fifo/msgfifo.go:51]: ok, I will split it to another pr.

**Pull request merged**